woodwork/column_accessor.py

```diff
@@ -184,6 +185,25 @@ def reset_semantic_tags(self):
         self._schema['semantic_tags'] = _reset_semantic_tags(self.logical_type.standard_tags,
                                                              self.use_standard_tags)
 
+    def set_logical_type(self, logical_type):
+        """Update the logical type for the series, clearing any previously set semantic tags,
+        and returning a new Series.
+
+        Args:
+            logical_type (LogicalType, str): The new logical type to set for the series.
+
+        Returns:
+            Series: A new series with the updated logical type.
+        """
+        logical_type = _get_column_logical_type(self._series, logical_type, self._series.name)
```

Comment:
`init_series` is going to just call `_get_column_logical_type` on the logical type. So not sure if this line is necessary.

And the fact that this can basically just be a wrapper for `init_series` makes me think even more that `set_logical_type` on the column isn't super important....

Reply:
Good catch. I'll remove the duplicate call here.

I agree on the usefulness of this method, but will still create an issue to review in more detail before we remove completely.